client/model/dependencyModel.js

```diff
@@ -105,12 +106,17 @@ define(['knockout', 'model/constants'], function(ko, constants) {
 
         self.requiredBy = ko.computed(function() {
             var dependencies = ko.observableArray([]);
+            // HATE HATE HATE this double loop. Need to move this processing to the server side.
```

Comment:
Probably a lot. Here goes...

upstream and downstream dependencies are already mapped on the server: /api/application/dependencies

Some ideas:
- modify the update callback for dependencies to parse this data
- rather than create the data structures/mappings in memory, do a separate API call for the dependency info when you expand the dependency dropdown (similar to how the server notes work)
- I'm not sure how the above would work with the other D3 views...that would need to be investigated